sasl.go

```diff
@@ -85,6 +86,37 @@ func ConnSASLAnonymous() ConnOption {
 	}
 }
 
+// ConnSASLExternal enables SASL EXTERNAL authentication for the
+// connection if secured by TLS and offered by server.
+func ConnSASLExternal() ConnOption {
+	return func(c *conn) error {
+		// make handlers map if no other mechanism has
+		if c.saslHandlers == nil {
+			c.saslHandlers = make(map[encoding.Symbol]stateFunc)
+		}
+
+		// add the handler the the map
+		c.saslHandlers[saslMechanismEXTERNAL] = func() stateFunc {
+			init := &frames.SASLInit{
+				Mechanism:       saslMechanismEXTERNAL,
+				InitialResponse: []byte(""),
```

Comment:
@HolyKau I can't find any docs on this, do you have a link?  How did you test it?

Reply:
The best docs I found for this was from [here](https://datatracker.ietf.org/doc/html/rfc4422#appendix-A).

This was tested by configuring a RabbitMQ server with the EXTERNAL authentication mechanism and making a TLS connection to it with authentication provided by self-signed certificates. RabbitMQ was running on a Kubernetes cluster and was deployed using [RabbitMQ Operator](https://www.rabbitmq.com/kubernetes/operator/quickstart-operator.html).

The test code looked something like this:
```
type publisher struct {
	conn      *amqp.Client
	session   *amqp.Session
	configs   *queueConfiguration.Config
	tlsConfig *tls.Config
}

type Publisher interface {
	messaging.Publisher
}

func NewPublisher(url string, tlsConfig *tls.Config, authType string) (Publisher, error) {
	var conn *amqp.Client
	var err error

	if tlsConfig != nil {
		cfg := amqp.ConnTLSConfig(tlsConfig)
		if authType == queueConfiguration.MutualTLSAuthType {
			conn, err = amqp.Dial(url, cfg, amqp.ConnSASLExternal())
		} else if authType == queueConfiguration.PlainTLSAuthType {
			conn, err = amqp.Dial(url, cfg)
		} else {
			return nil, fmt.Errorf("Invalid auth type: %s", authType)
		}
	} else {
		conn, err = amqp.Dial(url)
	}

	if err != nil {
		return nil, err
	}

	session, err := conn.NewSession()
	if err != nil {
		return nil, fmt.Errorf("Channel: %s", err)
	}

	configs, _, _ := queueConfiguration.GetConfig()

	ret := &publisher{
		conn: conn,
		session: session,
		configs: configs,
	}

	return ret, nil
}
```

Reply:
Thanks this helps a lot.

I see there are two examples, one where the initial response is empty (like in your PR) and one where it isn't.  Do you think it makes sense to update `ConnSASLExternal()` to take a `string` param so that both cases can be covered?

Reply:
Ya, I agree it would make sense to pass in a `string` param

Reply:
Sounds good.

Please note that I rebased your branch on latest `master`.